mllib/src/main/scala/org/apache/spark/ml/param/sharedParams.scala

```diff
@@ -23,6 +23,27 @@ private[ml] trait HasRegParam extends Params {
   def getRegParam: Double = get(regParam)
 }
 
+private[ml] trait HasElasticNetParam extends HasRegParam {
+  /** param for elastic net regularization parameter */
+  val alphaParam: DoubleParam = new DoubleParam(
```

Comment:
LOR will have it eventually as well. Do you mean we should `with HasAlpha with HasRegParam` instead of `with HasElasticNetParam`?